sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -883,7 +895,7 @@ class Column(protected[sql] val expr: Expression) extends Logging {
    * @group expr_ops
    * @since 1.4.0
    */
-  def as(aliases: Seq[String]): Column = withExpr { MultiAlias(expr, aliases) }
+  def as(aliases: Seq[String]): Column = name(aliases)
```

Comment:
Sorry I didn't make it clearly: delegate to `def name(alias: String)` if we can. For this one, we don't need to touch it as it doesn't make sense to call in on `TypedColumn`.

The reason I want to make as more as possible alias methods delegate to  `def name(alias: String)` is: we don't need to override all of them in `TypedColumn`, but just the `name`.